modules/arrow/src/workers/triangulation-worker.ts

```diff
@@ -51,28 +51,42 @@ function triangulateBatch(data: TriangulateInput): TriangulateResult {
 /**
  * Reading the arrow file into memory is very fast. Parsing the geoarrow column is slow, and blocking the main thread.
  * To address this issue, we can move the parsing job from main thread to parallel web workers.
- * Each web worker will parse the geoarrow column using one chunk/batch of arrow data, and return binary geometries to main thread.
+ * Each web worker will parse one chunk/batch of geoarrow column, and return binary geometries to main thread.
  * The app on the main thread will render the binary geometries and the parsing will not block the main thread.
  *
  * @param data
  * @returns
  */
 function parseGeoArrowBatch(data: ParseGeoArrowInput): ParseGeoArrowResult {
-  let binaryGeometries: BinaryDataFromGeoArrow | null = null;
-  const {arrowData, chunkIndex, geometryColumnName, geometryEncoding, meanCenter, triangle} = data;
-  const arrowTable = arrow.tableFromIPC(arrowData);
-  const geometryColumn = arrowTable.getChild(geometryColumnName);
+  let binaryDataFromGeoArrow: BinaryDataFromGeoArrow | null = null;
+  const {chunkData, chunkIndex, geometryEncoding, meanCenter, triangle} = data;
+  // rebuild chunkData
+  const arrowData = new arrow.Data(
+    chunkData.type,
+    chunkData.offset,
+    chunkData.length,
+    chunkData.nullCount,
+    chunkData.buffers,
+    chunkData.children,
```

Comment:
Note that while many things here are pure data, I expect that `children` can contain class instances that are not transferrable to workers.